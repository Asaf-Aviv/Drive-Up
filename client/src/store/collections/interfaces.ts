import { MovieShowcase } from '../movies/interfaces';

export interface Collection {
  id: number;
  name: string;
  overview: string;
  poster_path: string | null;
  backdrop_path: string | null;
  parts: MovieShowcase[];
}
