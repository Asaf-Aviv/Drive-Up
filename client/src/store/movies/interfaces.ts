export interface Results<T> {
  results: T;
  page: number;
  total_pages: number;
  total_results: number;
}

export interface Genre {
  id: number;
  name: string;
}

export interface Series {
  id: number;
  name: string;
  poster_path: string;
  backdrop_path: string;
}

export interface Company {
  name: string;
  id: number;
  logo_path: string | null;
  origin_country: string;
}

export interface Image {
  file_path: string;
  width: number;
  height: number;
}

export interface Video {
  id: string;
  key: string;
  name: string;
  site: string;
  type: string;
}

export interface MovieShowcase {
  poster_path: string | null;
  adult: boolean;
  overview: string;
  release_date: string;
  genre_ids: number[];
  id: number;
  original_title: string;
  original_language: string;
  title: string;
  backdrop_path: string | null;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
}

export interface Movie {
  adult: boolean;
  backdrop_path: string | null;
  belongs_to_collection: Series | null;
  budget: number;
  genres: Genre[];
  homepage: string | null;
  id: number;
  images: {
    posters: Image[];
    backdrops: Image[];
  };
  videos: {
    results: Video[];
  };
  similar: Results<MovieShowcase[]>;
  recommendations: Results<MovieShowcase[]>;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string | null;
  production_companies: Company[];
  release_date: string;
  revenue: number;
  runtime: number | null;
  tagline: string | null;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}
