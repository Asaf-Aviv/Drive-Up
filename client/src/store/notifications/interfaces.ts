type NotificationVariant = 'error' | 'success' | 'info' | 'warning';

export interface Notification {
  id: string;
  variant: NotificationVariant;
  message: string;
}
