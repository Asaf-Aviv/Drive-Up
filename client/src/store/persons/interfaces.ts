export interface PersonSummary {
  profile_path: string;
  adult: boolean;
  id: number;
  name: string;
  popularity: number;
  known_for_department: string;
  gender: 0 | 1 | 2;
}
