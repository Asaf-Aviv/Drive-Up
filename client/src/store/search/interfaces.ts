import { MovieShowcase } from '../movies/interfaces';
import { PersonSummary } from '../persons/interfaces';
import { ShowShowcase } from '../shows/interfaces';

interface MediaType<T = 'movie' | 'tv' | 'person'> {
  media_type: T;
}

export type SearchResults =
  | MovieShowcase & MediaType<'movie'>
  | ShowShowcase & MediaType<'tv'>
  | PersonSummary & MediaType<'person'>;
