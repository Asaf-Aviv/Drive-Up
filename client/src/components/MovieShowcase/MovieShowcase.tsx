import React from 'react';
import { Link } from 'react-router-dom';
import { MovieShowcase as IMovieShowcase } from '../../store/movies/interfaces';

const MovieShowcase: React.FC<IMovieShowcase> = ({
  id,
  title,
  original_language,
  overview,
  poster_path,
}) => (
  <Link to={`/movies/${id}`}>
    <div>
      <h2>{title}</h2>
      <p>{original_language}</p>
      <p>{overview}</p>
      <img
        src={`http://image.tmdb.org/t/p/w92/${poster_path}`}
        alt={title}
      />
    </div>
  </Link>
);

export default MovieShowcase;
