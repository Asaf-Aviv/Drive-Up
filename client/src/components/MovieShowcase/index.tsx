import MovieShowcase from './MovieShowcase';

export default MovieShowcase;
