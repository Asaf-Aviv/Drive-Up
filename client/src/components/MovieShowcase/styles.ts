import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  paper: {
    marginBottom: 24,
    width: '100%',
  },
  link: {
    textDecoration: 'none',
    color: 'inherit',
  },
  buttonLink: {
    textDecoration: 'none',
    marginRight: 16,
  },
  footerLink: {
    display: 'inline-block',
    marginLeft: 'auto',
    width: 150,
    '&:hover': {
      '& .MuiTypography-body1': {
        marginRight: 16,
      },
    },
    '& .MuiTypography-body1': {
      transition: 'margin-right 150ms',
      marginRight: 8,
    },
  },
  overview: {
    maxWidth: 650,
    textAlign: 'center',
    fontSize: '0.9em',
    flex: 1,
  },
});

export default useStyles;
