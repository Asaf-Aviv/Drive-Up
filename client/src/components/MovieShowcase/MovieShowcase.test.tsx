import { mount } from 'enzyme';
import MovieShowcase from './MovieShowcase';
import { MovieShowcase as IMovieShowcase } from '../../store/movies/interfaces';
import { renderWithProviders } from '../../test-utils';

const movie: IMovieShowcase = {
  adult: false,
  backdrop_path: '/4W0FnjSGn4x0mKZlBRx8OjFxQUM.jpg',
  genre_ids: [27],
  id: 474350,
  original_language: 'en',
  original_title: 'It Chapter Two',
  overview: '27 years after',
  popularity: 450.453,
  poster_path: '/zfE0R94v1E8cuKAerbskfD3VfUt.jpg',
  release_date: '2019-09-06',
  title: 'It Chapter Two',
  video: false,
  vote_average: 7.4,
  vote_count: 135,
};

describe('MovieShowcase component', () => {
  it('renders correctly', () => {
    const wrapper = mount(renderWithProviders(MovieShowcase, movie));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render the title of the movie', () => {
    const wrapper = mount(renderWithProviders(MovieShowcase, movie));
    const title = wrapper.find('h2').text();
    expect(title).toEqual(movie.title);
  });

  it('should render the overview of the movie', () => {
    const wrapper = mount(renderWithProviders(MovieShowcase, movie));
    const title = wrapper.find('h6').text();
    expect(title).toEqual(movie.overview);
  });

  it('should render a link to the movie by id', () => {
    const wrapper = mount(renderWithProviders(MovieShowcase, movie));
    const href = wrapper.find('a').prop('href');
    expect(href).toEqual(`/movie/${movie.id}`);
  });

  it('should render the movie image', () => {
    const wrapper = mount(renderWithProviders(MovieShowcase, movie));
    const imgSrc = wrapper.find('img').prop('src');
    const baseSrc = 'http://image.tmdb.org/t/p/w';

    expect(wrapper.exists('img')).toBeTruthy();
    expect(imgSrc).toContain(baseSrc);
    expect(imgSrc).toContain(movie.poster_path);
  });
});
