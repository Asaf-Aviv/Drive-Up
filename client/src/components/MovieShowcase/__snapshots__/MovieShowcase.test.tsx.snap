// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieShowcase component renders correctly 1`] = `
"<WithStyles(ForwardRef(Paper)) className=\\"makeStyles-paper-1\\" elevation={5}>
  <Styled(MuiBox) display=\\"flex\\">
    <Styled(MuiBox) display=\\"inline-flex\\" flexBasis={154}>
      <img style={{...}} src=\\"http://image.tmdb.org/t/p/w154//zfE0R94v1E8cuKAerbskfD3VfUt.jpg\\" alt=\\"It Chapter Two\\" />
    </Styled(MuiBox)>
    <Styled(MuiBox) padding={3} width=\\"100%\\">
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <Styled(MuiBox) flex={1} component=\\"span\\" paddingRight={3} fontWeight={600}>
          Release Date: 9/6/2019
        </Styled(MuiBox)>
        <Link className=\\"makeStyles-link-2\\" to=\\"/movie/474350\\">
          <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\" align=\\"center\\" gutterBottom={true}>
            It Chapter Two
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Styled(MuiBox) flex={1} textAlign=\\"right\\" component=\\"span\\">
          Language EN
        </Styled(MuiBox)>
      </Styled(MuiBox)>
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <WithStyles(ForwardRef(Typography)) className=\\"makeStyles-overview-3\\" variant=\\"subtitle1\\">
          27 years after
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>"
`;
